Controller/AuthorizeController.php

```diff
@@ -28,66 +35,147 @@
  *
  * @author Chris Jones <leeked@gmail.com>
  */
-class AuthorizeController implements ContainerAwareInterface
+class AuthorizeController
 {
     /**
      * @var ClientInterface
      */
     private $client;
 
     /**
-     * @var ContainerInterface
+     * @var SessionInterface
      */
-    protected $container;
+    private $session;
 
     /**
-     * Sets the container.
+     * @var Form
+     */
+    private $authorizeForm;
+
+    /**
+     * @var AuthorizeFormHandler
+     */
+    private $authorizeFormHandler;
+
+    /**
+     * @var OAuth2
+     */
+    private $oAuth2Server;
+
+    /**
+     * @var EngineInterface
+     */
+    private $templating;
+
+    /**
+     * @var RequestStack
+     */
+    private $requestStack;
+
+    /**
+     * @var TokenStorageInterface
+     */
+    private $tokenStorage;
+
+    /**
+     * @var Router
+     */
+    private $router;
+
+    /**
+     * @var ClientManagerInterface
+     */
+    private $clientManager;
+
+    /**
+     * @var string
+     */
+    private $templateEngineType;
+
+    /**
+     * @var EventDispatcher
+     */
+    private $eventDispatcher;
+
+    /**
+     * This controller had been made as a service due to support symfony 4 where all* services are private by default.
+     * Thus, there is considered a bad practice to fetch services directly from container.
+     * @todo This controller could be refactored to do not rely on so many dependencies
      *
-     * @param ContainerInterface|null $container A ContainerInterface instance or null
+     * @param RequestStack           $requestStack
+     * @param SessionInterface       $session
+     * @param Form                   $authorizeForm
+     * @param AuthorizeFormHandler   $authorizeFormHandler
+     * @param OAuth2                 $oAuth2Server
+     * @param EngineInterface        $templating
+     * @param TokenStorageInterface  $tokenStorage
+     * @param Router                 $router
+     * @param ClientManagerInterface $clientManager
+     * @param EventDispatcher        $eventDispatcher
+     * @param string                 $templateEngineType
      */
-    public function setContainer(ContainerInterface $container = null)
```

Comment:
@dnahrebecki as commented earlier, please ensure you do NOT break BC in this PR, I'm planning on starting a 2.x branch rather soon and revamping this lib.